lib/handlers/router-entities/pool-caching/v3/dynamo-pool-provider.ts

```diff
@@ -0,0 +1,110 @@
+import { ChainId, ICache, IV3PoolProvider, V3PoolAccessor } from '@uniswap/smart-order-router'
+import { Token } from '@uniswap/sdk-core'
+import { ProviderConfig } from '@uniswap/smart-order-router/build/main/providers/provider'
+import { FeeAmount, Pool } from '@uniswap/v3-sdk'
+import { DynamoDB } from 'aws-sdk'
+
+export class DynamoPoolProvider implements IV3PoolProvider {
```

Comment:
@mikeki I find it a bit awkward that I'm copying and pasting a lot of code from [caching-pool-provider.ts](https://github.com/Uniswap/smart-order-router/blob/main/src/providers/v3/caching-pool-provider.ts), but I think it's better to minimize the refactoring in [smart-order-router](https://github.com/Uniswap/smart-order-router) right? Also in this case, we will be able to delete [caching-pool-provider.ts](https://github.com/Uniswap/smart-order-router/blob/main/src/providers/v3/caching-pool-provider.ts) upon migration to dynamo cache.

Reply:
I think you no longer are doing alot of copy pasting?